test/test_final.py

```diff
@@ -0,0 +1,25 @@
+from dataclasses import dataclass
+
+from .testutils import *
+
+
+@pytest.mark.skipif(
+    sys.version_info < (3, 8), reason="Final is only supported in Python >= 3.8"
```

Comment:
See previous comment: This test should work in all versions, but use the Final from `typing_extensions` if python < 3.8